controllers/authorino_controller.go

```diff
@@ -0,0 +1,598 @@
+			"Failed to get roleBinding %s for authorino instance %s, err: %d",
+			roleBinding.GetName(),
+			authorino.GetName(),
+			err,
+		)
+	}
+
+	if errors.IsNotFound(err) {
+		roleBinding.RoleRef = rbacv1.RoleRef{
+			Name: roleBinding.GetName(),
+			Kind: "Role",
+		}
+		roleBinding.Subjects = []rbacv1.Subject{
+			{
+				Kind:      rbacv1.ServiceAccountKind,
+				Name:      saName,
+				Namespace: authorino.GetNamespace(),
+			},
+		}
+		ctrl.SetControllerReference(authorino, roleBinding, r.Scheme)
+		err = r.Create(context.TODO(), roleBinding)
+		if err != nil {
+			return fmt.Errorf(
+				"Failed to create leader election role binding, err: %d",
+				err,
+			)
+		}
+	}
+	return nil
+}
+
+func getLeaderElectionRules() []rbacv1.PolicyRule {
+	return []rbacv1.PolicyRule{
+		{
+			APIGroups: []string{"*"},
+			Resources: []string{"configmaps"},
+			Verbs:     []string{"get", "list", "watch", "create", "update", "patch", "delete"},
+		},
+		{
+			APIGroups: []string{"*"},
+			Resources: []string{"configmaps/status"},
+			Verbs:     []string{"get", "update", "patch"},
+		},
+		{
+			APIGroups: []string{"*"},
+			Resources: []string{"events"},
+			Verbs:     []string{"create", "patch"},
+		},
+		{
+			APIGroups: []string{"coordination.k8s.io"},
+			Resources: []string{"leases"},
+			Verbs:     []string{"get", "list", "create", "update"},
+		},
+	}
+}
+
+func (r *AuthorinoReconciler) authorinoServices(authorino *authorinooperatorv1beta1.Authorino) error {
+
+	services := make(map[string][]corev1.ServicePort)
+	services["authorino-authorization"] = []corev1.ServicePort{
+		{
+			Name:     "grpc",
+			Port:     50051,
+			Protocol: corev1.ProtocolTCP,
+		},
+	}
+	services["authorino-oidc"] = []corev1.ServicePort{
+		{
+			Name:     "http",
+			Port:     8083,
+			Protocol: corev1.ProtocolTCP,
+		},
+	}
+	services["authorino-controller-manager-metrics-service"] = []corev1.ServicePort{
+		{
+			Name:       "https",
+			Port:       8443,
+			TargetPort: intstr.FromString("https"),
+		},
+	}
+
+	for name, service := range services {
+		obj := &corev1.Service{
+			ObjectMeta: v1.ObjectMeta{
+				Name:      authorino.GetName() + "-" + name,
+				Namespace: authorino.GetNamespace(),
+			},
+		}
+
+		ctrl.SetControllerReference(authorino, obj, r.Scheme)
+
+		_, err := controllerutil.CreateOrUpdate(context.TODO(), r.Client, obj, func() error {
+			obj.Spec.Ports = service
+			obj.Spec.Selector = labelsForAuthorino(authorino.GetName())
+			return nil
+		})
+		if err != nil {
+			return fmt.Errorf("Failed creating %s service, err: %w", name, err)
+		}
+	}
+
+	return nil
+}
+
+func (r *AuthorinoReconciler) handleStatusUpdate(authorino *authorinooperatorv1beta1.Authorino) error {
+	authorinoInstances := authorinooperatorv1beta1.AuthorinoList{}
+	err := r.List(context.TODO(), &authorinoInstances)
+	if err != nil {
+		return fmt.Errorf("Failed getting authorino instances, err: %w", err)
+	}
+
+	authorino.Status.AuthorinoInstances = len(authorinoInstances.Items)
```

Comment:
I don't get this field. Why would an Authorino instance hold the count of how many Authorino instances there are running in the space?

Reply:
yeah, it doesn't make any sense I was trying to fill up the status block with more things, but as I mentioned above I will create an issue to improve the handling of the status block using conditions. 

I will remove this field.